docs/config/lua/wezterm.table/merge.md

```diff
@@ -0,0 +1,36 @@
+# `wezterm.table.merge(array_of_tables [, keep_first])`
+
+{{since('nightly')}}
+
+This function merges Lua tables passed to it in the form of an array.
+I.e., to merge the Lua tables `tbl1` and `tbl2`, we can pass them to
+the function as `{ tbl1, tbl2 }`. (See below.)
+
+By default this function merges tables with identical keys by taking
+the value from the last table in the array with each given key.
+
+The optional `keep_first` allows us to instead prefer values from the
+first table in the array where we see the key by passing `true` after the array.
+The default behavior is identical to what we get by passing `false`.
+
+```lua
+local wezterm = require 'wezterm'
+local merge = wezterm.table.merge
+
+local tbl1 = {
+  a = 1,
+  b = '2',
+}
+
+local tbl2 = {
+  a = '1',
+  c = 3,
+}
+
+wezterm.log_error(merge { tbl1, tbl2 })
```

Comment:
I think it would be helpful to understand the result of this function if you could illustrate it with an assertion:

```lua
assert(merge{tbl1, tbl2}, {a='1', b='2', c=3})
```

Reply:
Note: the sample assert suggested by @wez is not valid / doesn't do what you expect. Since there are no built-in table comparison functions in Lua we could just use a fake `assert_equal` function in docs like this, even if that function doesn't actually exist?

Reply:
ah, then maybe:

```lua
assert(to_string(merge{tbl1, tbl2}), "{a='1', b='2', c=3}")
```

Reply:
> ah, then maybe:
> 
> ```lua
> assert(to_string(merge{tbl1, tbl2}), "{a='1', b='2', c=3}")
> ```

We can also do something like this, but it should wait until we decide what to do with `to_string`.

Reply:
I added `wezterm.table.equal( tbl1, tbl2 )` so we can use that in the `assert`.